Remove superfluous deserializePart overload

Geod24 opened the pull request:
It is not necessary anymore, since it doesn't do anything in place.
The whole distinction between deserializePart / deserializeFull is also moot,
and should be removed in a later commit.

Geod24 committed:
Remove superfluous deserializePart overload

It is not necessary anymore, since it doesn't do anything in place.
The whole distinction between deserializePart / deserializeFull is also moot,
and should be removed in a later commit.

TrustHenry commented on source/agora/common/Set.d:
```diff
@@ -98,29 +98,11 @@ public struct Set (T)
-                this._set[value] = true;
-            }
-        }
+            this._set[deserializeFull!T(dg)] = true;
```
```
unittest
{
    import agora.common.Set;
    import agora.common.Types;
    import agora.common.Deserializer;
    import agora.common.Serializer;
    import agora.consensus.data.Transaction;

    auto tx_set = Set!Transaction([Transaction.init]);
    auto bytes = tx_set.serializeFull();
    auto new_txs = deserializeFull!(Set!Transaction)(bytes);
    assert(tx_set == new_txs);
}
```

How about this test code?

Geod24 replied:
Added it, thanks!

Geod24 committed:
Work around DMD bug in testSymmetry

DMD complains about comparison of values with different qualifiers
(e.g. immutable vs mutable).
So when deserializing, just get a 'const' value (as 'value', the parameter, is const).

Geod24 commented:
Added @TrustHenry 's testcase, and a workaround for a slight bug it found.

TrustHenry reviewed (approved):
LGTM

linked0 commented:
LGTM

MichaelKim20 reviewed (approved):
LGTM
